docs/spec-files/ecdh1pu.md

```diff
@@ -0,0 +1,112 @@
+
+The final `Z` is the concatenation of `Ze` and `Zs` which is then used in the key wrapping the same way as in ECDH-ES.
+
+There are special considerations in the process to protect sender impersonation as described [2.1](https://datatracker.ietf.org/doc/html/draft-madden-jose-ecdh-1pu-04#section-2.1) of the draft:
+```
+   In Key Agreement with Key Wrapping mode, the JWE Authentication Tag
+   is included in the input to the Key Derivation Function as described
+   in section Section 2.3.  This ensures that the content of the JWE was
+   produced by the original sender and not by another recipient, as
+   described in section Section 4.
+
+   Key Agreement with Key Wrapping mode MUST only be used with content
+   encryption algorithms that are compactly committing AEADs as
+   described in [ccAEAD].  The AES_CBC_HMAC_SHA2 algorithms described in
+   section 5.2 of [RFC7518] are compactly committing and can be used
+   with ECDH-1PU in Key Agreement with Key Wrapping mode.  Other content
+   encryption algorithms MUST be rejected.  In Direct Key Agreement
+   mode, any JWE content encryption algorithm MAY be used.
+
+   The requirement to include the JWE Authentication Tag in the input to
+   the Key Derivation Function implies an adjustment to the order of
+   operations performed during JWE Message Encryption described in
+   section 5.1 of [RFC7516].  Steps 3-8 are deferred until after step
+   15, using the randomly generated CEK from step 2 for encryption of
+   the message content.
+```
+
+To sum up, these considerations require:
+1. The use of the `AES_CBC_HMAC_SHA` family of content encryption algorithms to encrypt the payload. Currently, JWE supports the following three algorithms in this family:
+   1. A128CBC-HS256
+   2. A192CBC-HS384
+   3. A256CBC-HS512
+   
+    But the spec only mandates [iii](https://identity.foundation/didcomm-messaging/spec/#curves-and-content-encryption-algorithms) as required to minimize interoperability issues in the protocol.
+2. Encrypt the payload prior to wrapping `cek` with the derived `Z`. The output is `ciphertext` and `tag`
+3. Use the resulting `tag` in item 2 above as the value `len(tag)`+`tag` set in `cctag` in the keys derivations of `Ze` and `Zs`.
+
+
+Additionally, the `skid` protected header is also introduced as a kid to reference the sender key. This will help recipients resolve the key behind `skid` and execute the ECDH-1PU process explained in the previous section. 
+
+For the sake of consistency, `apu` and `apv` must be set to the values mentioned in [section 5.8](https://identity.foundation/didcomm-messaging/spec/#ecdh-1pu-key-wrapping-and-common-protected-headers) of the DIDComm Messaging protocol to further restrain and protect the message.
```

Comment:
I think this line is specific to DIDComm. Not sure if this is needed if the article is only about 1PU

Reply:
They're add here only as a hint to refer the spec.